presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -473,6 +478,23 @@ else if (parameterType.isPrimitive() && !hasNullFlag) {
             }
         }
 
+        private void checkTypeParameters(TypeSignature signature, Method method, ImmutableSet<String> typeParameterNames)
+        {
+            if (typeParameterNames.contains(signature.getBase())) {
+                checkArgument(signature.getParameters().isEmpty(), "Annotated type parameter %s on the method [%s] cannot take parameters.", signature.getBase(), method);
```

Comment:
Write message like `Expected type parameter to not take parameters, but got %s on method [%s].`